src/wincapture.cpp

```diff
@@ -268,6 +268,17 @@ namespace ros_win_camera
         spMediaType.copy_to(ppMediaType);
         return bMatch;
 
+    }
+    void WindowsMFCapture::GetCaptureConfig(uint32_t& width, uint32_t& height, float& framerate, GUID& videoSubtype)
+    {
```

Comment:
what is being guarded by         std::lock_guard g(m_apiGuardMutex);

Reply:
The guard is to avoid simultaneous  calls to/changes to SourceReader and resolution etc.